packages/thyra/core/test/operator_vector/DefaultProductVectorSpace_UnitTests.cpp

```diff
@@ -444,10 +446,11 @@ TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( DefaultProductVector,
   const RCP<const VectorSpaceBase<Scalar> > vs =
     createProductVectorSpace<Scalar>(g_localDim, numBlocks);
 
-  const RCP<const VectorBase<Scalar> > v = createMember(vs);
+  const RCP<VectorBase<Scalar> > v = createMember(vs);
+  Thyra::put_scalar<Scalar>(0.0, v.ptr());
 
   const RCP<const ProductVectorBase<Scalar> > prod_v =
-    castOrCreateProductVectorBase(v);
+    castOrCreateProductVectorBase(Teuchos::rcp_const_cast<const VectorBase<Scalar> >(v));
```

Comment:
FYI, as mentioned in section "5.7.3 Implicit type conversion problems and shortcomings" and subsection "Implicit conversions failing due to templated function" in [Teuchos C++ Memory Management Classes, Idioms, and Related Topics: The Complete Reference (A Comprehensive Strategy for Safe and Efficient Memory Management in C++ for High Performance Computing)](https://bartlettroscoe.github.io/publications/TeuchosMemoryManagementSAND.pdf), a cleaner way to allow this implicit conversion is to just add the template argument as `castOrCreateProductVectorBase<Scalar>()`.
```suggestion
    castOrCreateProductVectorBase<Scalar>(v);
```